src/vaults/restaking/UltraLRT.sol

```diff
@@ -645,4 +645,51 @@ contract UltraLRT is
     function getRate() external view returns (uint256) {
         return convertToAssets(10 ** decimals());
     }
+
+    function migrateToV2(UltraLRT _newVault, bytes calldata users) external onlyGovernance {
+        // check the governance
+        if (governance != _newVault.governance()) revert ReStakingErrors.InvalidGovernance();
```

Comment:
can we factor out these checks at initiation level?

Reply:
like you could probably event store a `migrationReady()` function that we run once